File: app/src/main/java/com/azure/minitwitter/ui/BottomModalTweetFragment.java
package com.azure.minitwitter.ui;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.BottomSheetDialogFragment;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import com.azure.minitwitter.R;
import com.azure.minitwitter.common.Constants;
import com.azure.minitwitter.data.TweetViewModel;

public class BottomModalTweetFragment extends BottomSheetDialogFragment {

    private TweetViewModel tweetViewModel;
    private int idTweetToDelete;

    public static BottomModalTweetFragment newInstance(int idTweet) {
        BottomModalTweetFragment fragment = new BottomModalTweetFragment();
        Bundle args = new Bundle();
        args.putInt(Constants.ARG_TWEET_ID,idTweet);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState){
        super.onCreate(savedInstanceState);

        if(getArguments() != null){
            idTweetToDelete = getArguments().getInt(Constants.ARG_TWEET_ID);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.bottom_modal_tweet_fragment, container, false);

        final NavigationView nav = v.findViewById(R.id.navigation_view_bottom_tweet);
        nav.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                int id = menuItem.getItemId();
                boolean res;

                switch (id){
                    case R.id.action_delete_tweet:
                        tweetViewModel.deleteTweet(idTweetToDelete);
                        getDialog().dismiss();
                        res = true;
                        break;
                        default:
                            res = false;
                }

                return res;
            }
        });

        return v;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        tweetViewModel = ViewModelProviders.of(getActivity()).get(TweetViewModel.class);
    }

}

File: app/src/main/java/com/azure/minitwitter/ui/tweets/BottomModalTweetFragment.kt
package com.azure.minitwitter.ui.tweets

import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.design.widget.BottomSheetDialogFragment
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import com.azure.minitwitter.R
import com.azure.minitwitter.common.Constants
import com.azure.minitwitter.data.TweetViewModel
import kotlinx.android.synthetic.main.bottom_modal_tweet_fragment.*

class BottomModalTweetFragment: BottomSheetDialogFragment(){

    private var idTweetToDelete: Int = -1
    private lateinit var tweetViewModel: TweetViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if(arguments != null){
            idTweetToDelete = arguments!!.getInt(Constants.ARG_TWEET_ID)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.bottom_modal_tweet_fragment, container, false)

        navigation_view_bottom_tweet.setNavigationItemSelectedListener { menuItem: MenuItem ->
            val id = menuItem.itemId
            var res: Boolean

            when(id){
                R.id.action_delete_tweet -> {
                    tweetViewModel.deleteTweet(idTweetToDelete)
                    dialog.dismiss()
                    res = true
                }
                else -> res = false

            }

            return res
        }

        return view
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        tweetViewModel = ViewModelProviders.of(activity!!).get(TweetViewModel::class.java)
    }

}